### PR TITLE
Port category-table and category-table-p to rust

### PR DESCRIPTION
**Pull request by shanavas786**

### CODE REVIEW

File: rust_src/src/category.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//! Routines to deal with category tables.
+
+use remacs_macros::lisp_fn;
+use remacs_sys::Qcategory_table;
+use lisp::LispObject;
+use threads::ThreadState;
+
+/// Return t if ARG is a category table.
+#[lisp_fn]
+fn category_table_p(obj: LispObject) -> LispObject {
```
Comment (birkenfeld) [inline rust_src/src/category.rs]: Argument name should be `arg`.

File: rust_src/src/category.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    LispObject::from_bool(
+        obj.is_char_table() && obj.eq(unsafe { LispObject::from_raw(Qcategory_table) }),
```
Comment (birkenfeld) [inline rust_src/src/category.rs]: Doesn't seem entirely correct, the original is checking arg->purpose

**Comment by shanavas786** [comment]
@birkenfeld Thanks updated.

**Comment by birkenfeld** [comment]
Nice!

**Comment by Wilfred** [comment]
Great stuff! :)

**Pull request merged**